Wrong library paths on macOS with Apple Silicon chip

Problem description
------------
On new architectures (Apple Silicon) Homebrew uses `/opt/homebrew/` as default installation directory instead of `/usr/local/`, which it uses on macOS Intel. See https://docs.brew.sh/Installation.

As a consequence, our library paths are wrong on Apple Silicon. This breaks the OpenMP support with the error
```python
epyccel/test_epyccel_openmp.py F

========================================================= FAILURES =========================================================
________________________________________________ test_directive_in_else[c] _________________________________________________

language = 'c'

    @pytest.mark.external
    def test_directive_in_else(language):
>       f1 = epyccel(openmp.directive_in_else, fflags = '-Wall', accelerators=['openmp'], language=language)

epyccel/test_epyccel_openmp.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../pyccel/epyccel.py:376: in epyccel
    mod, fun = epyccel_seq( python_function_or_module, **kwargs )
../pyccel/epyccel.py:225: in epyccel_seq
    execute_pyccel(pymod_filename,
../pyccel/codegen/pipeline.py:372: in execute_pyccel
    generated_filepath = create_shared_library(codegen,
../pyccel/codegen/python_wrapper.py:197: in create_shared_library
    sharedlib_filepath = src_compiler.compile_shared_library(wrapper_compile_obj,
../pyccel/codegen/compiling/compilers.py:458: in compile_shared_library
    self.run_command(cmd, verbose)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cmd = ['/usr/bin/gcc', '-shared', '-O3', '-funroll-loops', '-fPIC', '-Xpreprocessor', ...], verbose = False

    @staticmethod
    def run_command(cmd, verbose):
        """
        Run the provided command and collect the output.
    
        Run the provided compilation command, collect the output and raise any
        necessary errors if the file does not compile.
    
        Parameters
        ----------
        cmd : list of str
            The command to run.
        verbose : bool
            Indicates whether additional output should be shown.
    
        Returns
        -------
        str
            The exact command that was run.
    
        Raises
        ------
        RuntimeError
            Raises `RuntimeError` if the file does not compile.
        """
        cmd = [os.path.expandvars(c) for c in cmd]
        if verbose:
            print(' '.join(cmd))
    
        with subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                universal_newlines=True) as p:
            out, err = p.communicate()
    
        if verbose and out:
            print(out)
        if p.returncode != 0:
            err_msg = "Failed to build module"
            err_msg += "\n" + err
>           raise RuntimeError(err_msg)
E           RuntimeError: Failed to build module
E           clang: error: no such file or directory: '/usr/local/opt/libomp/lib'

../pyccel/codegen/compiling/compilers.py:500: RuntimeError
--------------------------------------------------- Captured stdout call ---------------------------------------------------

ERROR at shared library generation stage
------------------------------------------------- Captured stdout teardown -------------------------------------------------
Tearing down!
```

Diagnosis
--------
The following parameters are hardcoded in our compiler information, regardless of the actual installation directory of Homebrew:
```
"openmp": {
        "libdirs": [
            "/usr/local/opt/libomp/lib"
        ],
        "includes": [
            "/usr/local/opt/libomp/include"
        ]
    },
```

Suggested fix
--------
Get the correct paths from the following environment variables:
```shell
HOMEBREW_PREFIX
HOMEBREW_CELLAR
HOMEBREW_REPOSITORY
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,7 +40,7 @@ All notable changes to this project will be documented in this file.
 -   #1507 : Fix problems with name collisions in class functions.
 -   Ensure `pyccel-init` calls the related function.
 -   Stop unnecessarily importing deprecated NumPy classes `int`, `bool`, `float`, `complex` in Python translation.
--   #1713 : Get Homebrew directory on macOS with `brew --prefix`.
+-   #1712 : Fix library path and OpenMP support for recent Apple chips by getting Homebrew directory with `brew --prefix`.
 
 ### Changed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,6 +40,7 @@ All notable changes to this project will be documented in this file.
 -   #1507 : Fix problems with name collisions in class functions.
 -   Ensure `pyccel-init` calls the related function.
 -   Stop unnecessarily importing deprecated NumPy classes `int`, `bool`, `float`, `complex` in Python translation.
+-   #1713 : Get Homebrew directory on macOS with `brew --prefix`.
 
 ### Changed
 

--- a/pyccel/compilers/default_compilers.py
+++ b/pyccel/compilers/default_compilers.py
@@ -114,11 +114,11 @@ gcc_info = {'exec' : 'gcc',
             }
 
 if sys.platform == "darwin":
-    p = subprocess.run(['brew', '--prefix'], check=True, capture_output=True)
+    p = subprocess.run([shutil.which('brew'), '--prefix'], check=True, capture_output=True)
     HOMEBREW_PREFIX = p.stdout.decode().strip()
     OMP_PATH = os.path.join(HOMEBREW_PREFIX, 'opt/libomp')
 
-    gcc_info['openmp']['flags']    = ("-Xpreprocessor",'-fopenmp')
+    gcc_info['openmp']['flags']    = ("-Xpreprocessor", '-fopenmp')
     gcc_info['openmp']['libs']     = ('omp',)
     gcc_info['openmp']['libdirs']  = (os.path.join(OMP_PATH, 'lib'),)
     gcc_info['openmp']['includes'] = (os.path.join(OMP_PATH, 'include'),)

--- a/pyccel/compilers/default_compilers.py
+++ b/pyccel/compilers/default_compilers.py
@@ -8,9 +8,12 @@ import os
 import sys
 import sysconfig
 import subprocess
+import shutil
 
 from numpy import get_include as get_numpy_include
+
 from pyccel import __version__ as pyccel_version
+
 
 gfort_info = {'exec' : 'gfortran',
               'mpi_exec' : 'mpif90',
@@ -31,6 +34,7 @@ gfort_info = {'exec' : 'gfortran',
                   },
               'family': 'GNU',
               }
+
 if sys.platform == "win32":
     gfort_info['mpi_exec'] = 'gfortran'
     gfort_info['mpi']['flags']    = ('-D','USE_MPI_MODULE')

--- a/pyccel/compilers/default_compilers.py
+++ b/pyccel/compilers/default_compilers.py
@@ -111,10 +111,11 @@ gcc_info = {'exec' : 'gcc',
             'family': 'GNU',
             }
 if sys.platform == "darwin":
-    gcc_info['openmp']['flags'] = ("-Xpreprocessor",'-fopenmp')
-    gcc_info['openmp']['libs'] = ('omp',)
-    gcc_info['openmp']['libdirs'] = ('/usr/local/opt/libomp/lib',)
-    gcc_info['openmp']['includes'] = ('/usr/local/opt/libomp/include',)
+    OMP_PATH = os.path.join(os.environ['HOMEBREW_PREFIX'], 'opt/libomp')
+    gcc_info['openmp']['flags']    = ("-Xpreprocessor",'-fopenmp')
+    gcc_info['openmp']['libs']     = ('omp',)
+    gcc_info['openmp']['libdirs']  = (os.path.join(OMP_PATH, 'lib'),)
+    gcc_info['openmp']['includes'] = (os.path.join(OMP_PATH, 'include'),)
 elif sys.platform == "win32":
     gcc_info['mpi_exec'] = 'gcc'
     gcc_info['mpi']['flags']    = ('-D','USE_MPI_MODULE')

--- a/pyccel/compilers/default_compilers.py
+++ b/pyccel/compilers/default_compilers.py
@@ -7,6 +7,8 @@ import glob
 import os
 import sys
 import sysconfig
+import subprocess
+
 from numpy import get_include as get_numpy_include
 from pyccel import __version__ as pyccel_version
 
@@ -110,13 +112,19 @@ gcc_info = {'exec' : 'gcc',
                 },
             'family': 'GNU',
             }
+
 if sys.platform == "darwin":
-    OMP_PATH = os.path.join(os.environ['HOMEBREW_PREFIX'], 'opt/libomp')
+    p = subprocess.run(['brew', '--prefix'], check=True, capture_output=True)
+    HOMEBREW_PREFIX = p.stdout.decode().strip()
+    OMP_PATH = os.path.join(HOMEBREW_PREFIX, 'opt/libomp')
+
     gcc_info['openmp']['flags']    = ("-Xpreprocessor",'-fopenmp')
     gcc_info['openmp']['libs']     = ('omp',)
     gcc_info['openmp']['libdirs']  = (os.path.join(OMP_PATH, 'lib'),)
     gcc_info['openmp']['includes'] = (os.path.join(OMP_PATH, 'include'),)
+
 elif sys.platform == "win32":
+
     gcc_info['mpi_exec'] = 'gcc'
     gcc_info['mpi']['flags']    = ('-D','USE_MPI_MODULE')
     gcc_info['mpi']['libs']     = ('msmpi',)

